sklearn/utils/validation.py

```diff
@@ -83,16 +90,19 @@ def as_float_array(X, copy=True, force_all_finite=True):
     force_all_finite : boolean (default=True)
         Whether to raise an error on np.inf and np.nan in X.
 
+    allow_nan : boolean (default=False)
+        Whether to allow nan values in X.
```

Comment:
But I think you don't currently check in the case that `force_all_finite=False` and `allow_nan=False`. Do one of:
* handle this case
* document this behaviour
* make 'allow_nan' a value for `force_all_finite` rather than an additional parameter.